preprocessors/openstreetmap/osm_service.py

```diff
@@ -96,49 +101,60 @@ def get_timestamp():
     return timestamp
 
 
-def process_streets_data(OSM_data):
+def process_streets_data(OSM_data, bbox_coordinates):
     """Retrieve inteterested street information from the requested OSM data"""
     try:
         processed_OSM_data = []
+        lat_min = bbox_coordinates[0]
+        lat_max = bbox_coordinates[2]
+        lon_min = bbox_coordinates[1]
+        lon_max = bbox_coordinates[3]
         for way in OSM_data.ways:
             node_list = []
             for node in way.nodes:
-                node_object = {
-                    "id": int(node.id),
-                    "lat": float(node.lat),
-                    "lon": float(node.lon),
+                # Extract only nodes within the boundary
+                if node.lat >= lat_min and node.lat <= lat_max:
+                    if node.lon >= lon_min and node.lon <= lon_max:
+                        node_object = {
+                            "id": int(node.id),
+                            "lat": float(node.lat),
+                            "lon": float(node.lon),
+                        }
+                        if node_object not in node_list:
+                            node_list.append(node_object)
+            # Check if the "node_list" for a way is not empty.
+            # Otherwise all its nodes are outside the boundary, so exclude the
+            # way.
+            if node_list:
+                # Convert lanes to integer if its value is not None
+                lanes = way.tags.get("lanes")
+                if lanes is not None:
+                    lanes = int(lanes)
+                else:
```

Comment:
Why are these `else` statements included? They don't seem to do anything.

Reply:
```<way id="119842495">
    <nd ref="1345230863"/>
    <nd ref="216192133"/>
    <nd ref="1283567868"/>
    <tag k="highway" v="secondary"/>
    <tag k="lanes" v="2"/>
    <tag k="maxspeed" v="50"/>
    <tag k="name" v="Rue du Pont"/>
    <tag k="surface" v="asphalt"/>
  </way>```
Way keys such as the lanes, oneway, etc. usually have their raw values in strings. Meanwhile, the lanes are meant to be in integers while the other in Boolean. The if-statement does the conversion. There are also very rare cases where the "lanes" or "oneway" key has no value (e.g., ```"lanes": " "```). Any attempt to make a conversion in that case could produce a Python "ValueError". So, the idea in such a case is to accept the "empty string" as it is, which is what the else-statement is doing. The key would eventually be removed in line 156, since it has no value.

Reply:
I still do not see what you are accomplishing. You're setting the variable to itself which is `None`, would it not be filtered out anyway without these extra lines?